### ISSUE TITLE
GSLB_Pool_Member_A and _AAAA: virtualServer unexpected behaviour

### ISSUE THREAD

**Issue by alonsocamaro** [F5Networks/f5-appsvcs-extension#327]
### Environment
 * Application Services Version: f5-appsvcs-3.21.0-4.noarch
 * BIG-IP Version: 15.1.0.2

### Summary

Hi

I find that [documentation's example](https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/declarations/non-http-services.html#using-gslb-features-in-a-declaration) and the description of "GSLB_Pool_Member_A" and "_AAAA" in the schema itself could explain better the use of the element virtualServer

```
                "members": [
                    {
                        "ratio": 10,
                        "server": {
                            "use": "/Common/Shared/testServer"
                        },
                        "virtualServer": "0"
                    }
                ],
```

If I try using an index, AS3 returns: "01070226:3: Pool Member 0 references a nonexistent Virtual Server"

See full example attached.

If instead I use the name in the virtualServer element:

```
        "members": [
          {
            "server": {
              "use": "/Common/Shared/onprem_default"
            },
            "virtualServer": "https_account_onprem_default",
            "ratio": 50
          }
        ],

```

It works and creates the expected GTM configuration:

```
gtm pool a Application/pool_account_f5bddemos_io {
    members {
        /Common/onprem_default:https_account_onprem_default {
            member-order 0
            ratio 50
        }
    }
    partition gslb_f5bddemos_io
}
```

I'm not sure if this is just a documentation issue or an actual bug.

IMHO virtualServer should be pointer to a virtualServer but the schema doesn't (it's just a string, no schema checking?) and the actual description of the schema suggests the string has a boolean behavior ... Very confusing altogether.

Any clarifications will be greatly appreciated.

Thanks

### Steps To Reproduce

See attached files AS3 declarations, the first one indicating the virtualServer by name (works) and the second one by number (doesn't work)

[bigip-gslb-update-config.using-name.json.txt](https://github.com/F5Networks/f5-appsvcs-extension/files/5095465/bigip-gslb-update-config.using-name.json.txt)
[bigip-gslb-update-config.using-number.json.txt](https://github.com/F5Networks/f5-appsvcs-extension/files/5095466/bigip-gslb-update-config.using-number.json.txt)

**Comment by dstokesf5** [comment]
The virtualServer property is a string because you are selecting one of the virtual servers from the GSLB Server that is being referenced by a pointer. I do agree that the description is unclear, and could be improved.

Now, on to the number versus name issue. This is a result of a feature evolving over time, and the documentation not evolving well with it. Originally, you could not specify names on the GSLB Server's virtual servers. They were simply given a name based on their index in the list of virtual servers. We later added the name property to allow users to overwrite this behavior. When referencing a virtual server by number, you are really trying to reference it by its default name.

In short, reference by name, but use numbers if you did not specify names on your virtual servers.

**Comment by alonsocamaro** [comment]
@dstokesf5 could this be classified with the doc tag?

**Comment by dstokesf5** [comment]
I have added AUTOTOOL-1892 to our internal product backlog to improve the documentation around this functionality.

**Comment by sunitharonan** [comment]
It has been resolved.
